### PR TITLE
ISSUE-173 Changed way to add .ssh dir to get rid of bad owner permissions

### PR DESCRIPTION
**Pull request by juanmatias**
## what
* The way the `.ssh` dir is mounted into the container was changed from
  * mount ~/.ssh -> ~/.ssh
* to:
  * mount ~/.ssh -> /tmp/.ssh
  * cp -r /tmp/.ssh ~/
  * chown to container user and chmod to 600

## why
* If the `.ssh` files are owned by other than the current container user, or the config file has permissions other than `600` the `bad owner permissions issue` is got.

## references
* solves https://github.com/binbashar/leverage/issues/173

## Before release
* Review the checklist [here](https://binbash.atlassian.net/l/cp/BthxSQ0j)

### CODE REVIEW

**Comment by juanmatias** [comment]
IMPORTANT: do not merge this PR without documentation stating this new procedure

File: leverage/container.py [hunk 1]
```diff
@@ -417,15 +421,17 @@ def __init__(self, client):
             "AWS_CACHE_DIR": f"{self.guest_aws_credentials_dir}/cache",
             "SSO_CACHE_DIR": f"{self.guest_aws_credentials_dir}/sso/cache",
             "SCRIPT_LOG_LEVEL": get_script_log_level(),
-            "MFA_SCRIPT_LOG_LEVEL": get_script_log_level() # Legacy
+            "MFA_SCRIPT_LOG_LEVEL": get_script_log_level(), # Legacy
+            "SSH_AUTH_SOCK": '/ssh-agent' if not SSH_AUTH_SOCK is None else ""
```
Comment (diego-ojeda-binbash) [inline leverage/container.py:425]: @juanmatias Any chance this conditional could be easier to read? That "if not is None" puzzles me.

Comment (juanmatias) [reply leverage/container.py:425]: sure @diego-ojeda-binbash , pls check it now.... it's still a one line if, but I got rid of the "not"

File: leverage/container.py [hunk 2]
```diff
@@ -422,7 +422,7 @@ def __init__(self, client):
             "SSO_CACHE_DIR": f"{self.guest_aws_credentials_dir}/sso/cache",
             "SCRIPT_LOG_LEVEL": get_script_log_level(),
             "MFA_SCRIPT_LOG_LEVEL": get_script_log_level(), # Legacy
-            "SSH_AUTH_SOCK": '/ssh-agent' if not SSH_AUTH_SOCK is None else ""
+            "SSH_AUTH_SOCK": '' if SSH_AUTH_SOCK is None else '/ssh-agent'
```
Comment (diego-ojeda-binbash) [inline leverage/container.py:425]: @juanmatias Shouldn't this be: `SSH_AUTH_SOCK if SSH_AUTH_SOCK else '/ssh-agent'`?
That way the default value would be `'/ssh-agent'` unless it is overridden by setting the `SSH_AUTH_SOCK` environment variable.

**Review by diego-ojeda-binbash — APPROVED** [review]

**Review by exequielrafaela — APPROVED** [review]

File: leverage/container.py [hunk 3]
```diff
@@ -417,15 +421,17 @@ def __init__(self, client):
...
         }
         self.entrypoint = self.TF_BINARY
         self.mounts = [
             Mount(source=self.root_dir.as_posix(), target=self.guest_base_path, type="bind"),
             Mount(source=self.host_aws_credentials_dir.as_posix(), target=self.guest_aws_credentials_dir, type="bind"),
-            Mount(source=(self.home / ".ssh").as_posix(), target="/root/.ssh", type="bind"),
             Mount(source=(self.home / ".gitconfig").as_posix(), target="/etc/gitconfig", type="bind")
         ]
+        if not SSH_AUTH_SOCK is None:
```
Comment (angelofenoglio) [inline leverage/container.py:433]:
```suggestion
        if SSH_AUTH_SOCK is not None:
```
Works as well and is a bit more "natural language" but is totally a non issue.

**Review by angelofenoglio — APPROVED** [review]

**Pull request merged**